libexec/early_constructor/early_constructor.c

```diff
@@ -0,0 +1,18 @@
+#include <stdlib.h>
+
+extern volatile void* eptr;
```

Comment:
`void *eptr`, but also the volatile needs to be on eptr, not what it points to (which perhaps is why you ended up needing to make it a global to thwart today's optimisers?)

That is, this should work:

```c
__attribute__((__constructor__(101)))
 static void
foo(void)
{
	void * volatile eptr;

	eptr = malloc(1);
	free(eptr);
}
```

Reply:
Indeed, fixed.